packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -236,6 +253,26 @@ export abstract class QueryEditorStore {
     this.exportState = val;
   }
 
+  setIsQueryLoaderOpen(val: boolean): void {
+    this.isQueryLoaderOpen = val;
+  }
+
+  setIsLoadingQueryInProgress(val: boolean): void {
+    this.isLoadingQueryInProgress = val;
+  }
+
+  setQueries(val: LightQuery[]): void {
+    this.queries = val;
+  }
+
+  setSearchText(val: string): void {
```

Comment:
why can't we do this the same way we do it in `ExistingQuerySetupState.loadQueries()`? We don't need to store these many fields in the state, we can store them as component state instead, bebcause the modal is opened up and then the 2 actions people can do is:
1. Load a new query that will reload the app
2. Cancel and close the modal

I don't see much point in persisting these states in the `QueryEditorState` like this.